app/helpers/generic_file_helper.rb

```diff
@@ -39,7 +39,19 @@ def download_path(*args)
       gf = GenericFile.find(item.id)
       return gf.doi_url if gf.doi_url.present?
     end
-    return sufia.download_path(*args)
+    if args[1].nil?
+      return "/files/" + item.id + "/" + URI::encode(item.label) unless item.label.nil?
+      return "/files/" + item.id + "/nofile"
+    else
+      path = "/files/" + item.id + "?"
+      i = 0
+      args[1].each do |key,value|
```

Comment:
The args[1] bit is intended to handle the thumbnail case: ```/downloads/<noid>?file=thumbnail```, where args[1] comes through as ```{:file=>"thumbnail"}```. I'll add a comment.